pkg/log/log.go

```diff
@@ -149,6 +158,82 @@ func (l *OsqueryLogAdapter) logInfoAboutUnrecognizedProcessLockingPidfile(p []by
 	level.Debug(l.logger).Log(append(processInfo, "msg", "detected non-osqueryd process using pidfile")...)
 }
 
+// runAndLogPs runs ps filtering on the given PID, and logs the output.
+func (l *OsqueryLogAdapter) runAndLogPs(pid int) {
+	if runtime.GOOS == "windows" {
+		return
+	}
+
+	pidStr := strconv.Itoa(pid)
+	cmd := exec.Command("ps", "-p", pidStr, "-o", "user,pid,ppid,pgid,stat,time,command")
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		level.Debug(l.logger).Log(
+			"msg", "error running ps on non-osqueryd process using pidfile",
+			"pid", pid,
+			"err", err,
+		)
+		return
+	}
+
+	level.Debug(l.logger).Log(
+		"msg", "ran ps on non-osqueryd process using pidfile",
+		"pid", pid,
+		"output", string(out),
+	)
+}
+
+// runAndLogLsofByPID runs lsof filtering on the given PID, and logs the output.
+func (l *OsqueryLogAdapter) runAndLogLsofByPID(pid int) {
+	if runtime.GOOS == "windows" {
+		return
+	}
+
+	pidStr := strconv.Itoa(pid)
+	cmd := exec.Command("lsof", "-R", "-p", pidStr)
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		level.Debug(l.logger).Log(
+			"msg", "error running lsof on non-osqueryd process using pidfile",
+			"pid", pid,
+			"err", err,
+		)
+		return
+	}
+
+	level.Debug(l.logger).Log(
+		"msg", "ran lsof on non-osqueryd process using pidfile",
+		"pid", pid,
+		"output", string(out),
+	)
+}
+
+// runAndLogLsofOnPidfile runs lsof filtering by the osquery pidfile, and logs
+// the output.
+func (l *OsqueryLogAdapter) runAndLogLsofOnPidfile() {
+	if runtime.GOOS == "windows" {
+		return
+	}
+
+	fullPidfile := filepath.Join("/private", l.rootDirectory, "osquery.pid")
```

Comment:
Why `/private`? I don't think you need that

Reply:
Oh weird, I assumed I needed it because that's how it shows up in the output, but I do not. I will remove.

Reply:
On macOS `/var` is a symlink to `/private/var` but you can access it via `/var`. So we shouldn't assume that link is always present.